Android/app/src/go/intra/split/retrier.go

```diff
@@ -234,7 +247,55 @@ func splitHello(hello []byte) ([]byte, []byte) {
 	if s > limit {
 		s = limit
 	}
-	return hello[:s], hello[s:]
+	splitLen = int16(s)
+	pkts = net.Buffers{hello[:s], hello[s:]}
+
+	if len(pkts[0]) > MIN_TLS_HELLO_LEN {
+		// todo: Replace the following TLS fragmentation logic with tlsfrag.StreamDialer
+		//
+		// TLS record layout from RFC 8446:
+		//   [RecordType:1B][Ver:2B][Len:2B][Data...]
+		// RecordType := ... | handshake(22) | ...
+		//        Ver := 0x0301 ("TLS 1.0") | 0x0302 ("TLS 1.1") | 0x0303 ("TLS 1.2")
+		//
+		// Now we have already TCP-splitted into pkts0 (len >= 6) and pkts1.
+		// We just need to deal with pkts0 and fragment it:
+		//
+		//   original:   pkts[0]=[Header][data0],
+		//               pkts[1]=[data1]
+		//   fragmented: pkts[0]=[Header]
+		//               pkts[1]=[data0_0],
+		//               pkts[2]=[Header],
+		//               pkts[3]=[data0_1],
+		//               pkts[4]=[data1]
+
+		h1 := make([]byte, 5)
+		copy(h1, pkts[0][:5])
+		payload := pkts[0][5:] // len(payload) > 1 is guaranteed
+
+		typ := h1[0]
+		ver := binary.BigEndian.Uint16(h1[1:3])
+		recordLen := binary.BigEndian.Uint16(h1[3:5])
+
+		if typ == TYPE_HANDSHAKE && int(recordLen) >= len(payload) &&
+			(ver == VERSION_TLS10 || ver == VERSION_TLS11 ||
+				ver == VERSION_TLS12 || ver == VERSION_TLS13) {
+			rest := pkts[1]
+			frag := uint16(1 + rand.Intn(len(payload)-1)) // 1 <= frag <= len(payload)-1
+
+			binary.BigEndian.PutUint16(h1[3:5], frag)
+			payload1 := payload[:frag]
+
+			h2 := make([]byte, 5)
+			copy(h2, h1)
+			binary.BigEndian.PutUint16(h2[3:5], recordLen-frag) // recordLen >= len(payload) > frag
+			payload2 := payload[frag:]
+
+			pkts = net.Buffers{h1, payload1, h2, payload2, rest}
```

Comment:
Why are you returning a net.Buffers instead of a slice of slices? You never use its methods.

Reply:
not anymore, refactored to call `w.Write` directly.